tests/util.h

```diff
@@ -58,6 +58,29 @@ static void serializePacket(T &packet, const QByteArray &xml)
     QCOMPARE(data, processedXml);
 }
 
+///
+/// Serializes data and compares it to multiple XML representations.
+/// It fails if no comparison succeeds.
+///
+template<class T>
+static void serializePacket(T &packet, const QList<QByteArray> &xmls)
+{
+    for (const auto &xml : xmls) {
+        auto processedXml = xml;
+        processedXml.replace(u'\'', u'"');
+
+        const auto data = packetToXml(packet);
+        qDebug() << "expect " << processedXml;
+        qDebug() << "writing" << data;
+
+        if (data == processedXml) {
+            return;
+        }
+    }
+
+    QFAIL("No XML data equals the serialized packet");
+}
+
```

Comment:
This seems to be a very special case, I suggest you move it into the file where it's currently used. If this is later used in other places we can think about making it available to all tests.

Apart from that:
 * the success case is not logged (as QCOMPARE would do if it succeeds)
 * the packet is serialized `n` times
 * should use QVector or alternatively variadic template arguments

Reply:
The success case is logged and the `QList` is a `QVector` now.

The new `serializePacket()` is handy in all cases where a data structure with an undefined order (such as `QHash` or `QMultiHash`) is serialized . So, I think it would be better to keep it in that file. But if you think that it will not occur in the future, I can move it to `tests/qxmppomemodata/tst_qxmppomemodata.cpp`.

> the packet is serialized n times

How could that be improved?

Reply:
Testing it this way is a bad approach as it doesn't scale: you need to check against n! possible permutations, so for `n >= 3` you'll probably need a code generator for all permutations. Not to talk about the case when there's a randomly ordered list of elements with randomly ordered subelements.
That's why I don't want to make it 'official' by putting it in the util.h header file.

> > the packet is serialized n times
>
> How could that be improved?

do `const auto data = packetToXml(packet);` once, do the comparison in the loop.

Reply:
Thanks. I moved it to `tst_qxmppomemodata.cpp` and the serialization is only done once now.